aws_crt_common_ws/aws-crt-c-flags/src/lib.rs

```diff
@@ -0,0 +1,469 @@
+    /// let content = "test content".to_string();
+    /// build_info.write_generated_file_to_output_path(&content, Path::new("include/aws/common/config.h"));
+    ///```
+    pub fn write_generated_file_to_output_path(
+        &mut self,
+        generated_content: &String,
+        to: &Path,
+    ) -> &mut CRTModuleBuildInfo {
+        let target_location: String = format!(
+            "{}/{}",
+            env::var_os("OUT_DIR").unwrap().to_str().unwrap(),
+            to.display()
+        );
+        let target_path = Path::new(target_location.as_str());
+        fs::create_dir_all(target_path.parent().unwrap())
+            .expect("Creation of output directory failed.");
+        fs::write(target_location, generated_content).expect("Writing generated file failed!");
+
+        self
+    }
+
+    /// Returns the underlying toolchain that will be used for the build.
+    pub fn get_toolchain(&self) -> &Build {
+        &self.build_toolchain
+    }
+
+    /// Adds the file at path to the build tree
+    pub fn add_file_to_build(&mut self, path: &Path) -> &mut CRTModuleBuildInfo {
+        self.build_toolchain.file(path);
+        self
+    }
+
+    /// Attempts to compile, `to_compile` and returns a result on whether or not it succeeded.
+    /// This is useful for testing compiler capabilities before including a file or flag in your build.
+    ///
+    /// # Arguments
+    ///
+    /// * `to_compile` - C code to attempt compilation of.
+    ///
+    /// # Examples
+    /// ```should_panic
+    /// use aws_crt_c_flags::{CRTModuleBuildInfo};
+    /// let mut build_info = CRTModuleBuildInfo::new("aws_crt_common_sys");
+    /// build_info.try_compile("int main() { return 0; }").expect("This should have compiled");
+    /// ```
+    pub fn try_compile(&self, to_compile: &str) -> core::result::Result<(), cc::Error> {
+        // try_compile prints linker stuff. We don't want that since this is just for testing the
+        // compiler capabilities. Suppress it for this scope.
+        let _suppress_stdout_cause_the_build_prints_linker_nonsense = Gag::stdout().unwrap();
+        let mut test_build = Build::new();
+        let output_location = format!(
+            "{}/compiler_checks",
+            env::var_os("OUT_DIR").unwrap().to_str().unwrap()
+        );
+        test_build.out_dir(&output_location);
+        fs::create_dir_all(&output_location).expect("creation of try compile directory failed");
+        let target_location = format!(
+            "{}/compiler_checks/check.c",
+            env::var_os("OUT_DIR").unwrap().to_str().unwrap()
+        );
+        fs::write(Path::new(&target_location.as_str()), to_compile).expect("File write failed");
+        test_build.file(&target_location);
+        let res = test_build.try_compile("test");
+        fs::remove_dir_all(&output_location).expect("Cleanup of try compile step failed!");
+        res
+    }
+
+    fn load_to_build(&mut self) {
```

Comment:
After reading it, isn't this just `configure_cc_toolchain` or similar?

Reply:
i honestly don't think that's any more descriptive or clear.